doc/SAI-Proposal-ACL-1.md

```diff
@@ -0,0 +1,362 @@
+
+Figure 2 shows the relationship between ACL GROUPs and various bind points, it also provides a typical use case of allowing ACL TABLEs and ACL GROUPs to coexist and be bound to various bind points. Example 3 below shows how to create an ACL group and bind this ACL group to a port. Example 4 below shows how to bind the same ACL group to multiple bind points. Example 5 shows that an ACL table part of an ACL group can also be applied individually to any other bind point. The SAI implementation should handle such complex but intuitive scenarios and simplify application logic.<br>
+
+![SAI acl group](figures/sai_aclgroups.png "Figure 2: group ID and ACL ID's relation with several binding points. ")
+__Figure 2: group ID and ACL ID's relation with several binding points.__
+
+## ACL Stages 
+On creation of any ACL table or ACL table group, it is mandatory to provide one MANDATORY_ON_CREATE attributes to "create_acl_table" and "create_acl_table_group":<br>
+    - SAI_ACL_TABLE_ATTR_STAGE<br>
+    - SAI_ACL_TABLE_GROUP_ATTR_ACL_STAGE<br>
+On the same lines to create an ACL table, host adapters can optionally use CREATE_ONLY attributes to validate and reserve (if valid) an ACL table to be applied for a set of binding points identified by the enum sai_acl_bind_point_t:<br>
+    - SAI_ACL_TABLE_ATTR_ACL_BIND_POINT_LIST<br>
+    - SAI_ACL_TABLE_GROUP_ATTR_ACL_BIND_POINT_LIST<br>
+
+Based on attribute to validate ACL table or ACL table group, allows a SAI implementation to explicitly validate the scope of match fields and actions that can be supported at various bind points. Based on these explicit attributes the scope of ACL stages is restricted to INGRESS and EGRESS (removing the PRE_L2 and POST_L3). Hence, this proposal introduces well defined binding points along with specific stage(s) of the logical pipeline where any ACL table or ACL table group can be applied.<br>
+
+A new enumeration is added to handle the types of bind points that are currently supported in the ACL specification. This enum currently does not support bridge instance model.<br>
+
+    typedef enum _sai_acl_bind_point_t
+    {
+        /** Port Bind Point */
+        SAI_ACL_BIND_POINT_PORT,
+    
+        /** LAG Bind Point */
+        SAI_ACL_BIND_POINT_LAG,
+    
+        /** VLAN Bind Point */
+        SAI_ACL_BIND_POINT_VLAN,
+    
+        /** RIF Bind Point */
+        SAI_ACL_BIND_POINT_ROUTER_INTF,
+    
+        /** Port Bind Point */
+        SAI_ACL_BIND_POINT_SWITCH
+    } sai_acl_bind_point_t;
+
+## Metadata Usage Model
```

Comment:
not sure if we need this section here. this seems to be an independent topic and I cannot find it mentioned in the introduction.